index.js

```diff
@@ -10,8 +10,42 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  const response = await fetch(COLORS);
+
+  // handle errors with response
+  if (!response.ok) {
+    throw new Error(`Error with response. Response status: ${response.status}`);
+  }
+
+  const colors = await response.json();
+
+  // filters
+  const result = colors.filter((color) => {
+    // all colors whose name contains (includes) string, case insensitive
+    if (name) {
+      return color.name.toLowerCase().includes(name.toLowerCase());
```

Comment:
Neat. Very concise. Only comment is to outsource this comparison to a function. If the requirement changed to be case-sensitive you would have to change the comparison in multiple places